### PR TITLE
Restructure the Config objects

### PR DESCRIPTION
**Pull request by zwoop**
This unifies all Records access, and also adds support for the convenient proxy.config access for string records.

I also cleaned up some of the error handling as well.

### CODE REVIEW

**Comment by JosiahWI** [comment]
The AuTest proxy_protocol failed.


```
     Process: Default: Failed
       Setting up : Copying '/home/jenkins/workspace/Github_Builds/autest/src/build/proxy-verifier-v2.10.1/linux-amd64' to '/tmp/sandbox/proxy_protocol/pp-out-client-4/bin' - Passed
       Setting up : Copying 'replay/proxy_protocol_out.replay.yaml' to '/tmp/sandbox/proxy_protocol/pp-out-client-4/proxy_protocol_out.replay.yaml' - Passed
       Setting up : Copying '/home/jenkins/workspace/Github_Builds/autest/src/tests/gold_tests/autest-site/../../tools/proxy-verifier/ssl/client.pem' to '/tmp/sandbox/proxy_protocol/pp-out-client-4/client.pem' - Passed
       Setting up : Copying '/home/jenkins/workspace/Github_Builds/autest/src/tests/gold_tests/autest-site/../../tools/proxy-verifier/ssl/ca.pem' to '/tmp/sandbox/proxy_protocol/pp-out-client-4/ca.pem' - Passed
       Test : Checking that ReturnCode == 0 - Failed
          Reason: Returned Value 1 != 0
       Time-Out : Process finishes within expected time - Passed
          Reason: Returned value: 5.0228705406188965 < 600.0
       file /tmp/sandbox/proxy_protocol/_output/6-tr-Default/stream.all.txt : Verify the client got a 200 response. - Failed
          Reason: Contents of /tmp/sandbox/proxy_protocol/_output/6-tr-Default/stream.all.txt did not contains expression: "HTTP/1.1 200 OK"
       file /tmp/sandbox/proxy_protocol/_output/6-tr-Default/stream.stdout.txt : There should be no Proxy Verifier violation errors. - Passed
          Reason: Contents of /tmp/sandbox/proxy_protocol/_output/6-tr-Default/stream.stdout.txt excludes expression
```

**Comment by JosiahWI** [comment]
[approve ci autest]

File: src/cripts/Bundles/Common.cc [hunk 1]
```diff
@@ -105,20 +105,18 @@ Common::via_header(const Cript::string_view &destination, const Cript::string_vi
 }
 
 Common &
-Common::set_config(const Cript::string_view name, const Cript::Records::ValueType value)
+Common::set_config(const Cript::string_view name, const Cript::Records::ValueType &value)
 {
-  Cript::Records rec(name);
+  auto rec = Cript::Records::lookup(name); // These should be loaded at startup
```
Comment (cmcfarlen) [inline src/cripts/Bundles/Common.cc:110]: need to check rec to make sure its valid

**Review by cmcfarlen — CHANGES_REQUESTED** [review]

File: src/cripts/Bundles/Common.cc (hunk 1 above)
Comment (zwoop) [reply src/cripts/Bundles/Common.cc:110]: That's actually done by the lookup() method. I can change this though, which seems sensible.

Comment (zwoop) [reply src/cripts/Bundles/Common.cc:110]: Done

File: include/cripts/Lulu.hpp [hunk 2]
```diff
@@ -42,6 +42,15 @@ namespace CFS = std::filesystem;           // Simplify the usage of std::filesys
 
 integer integer_helper(std::string_view sv);
 
+// Some convenience macrods
```
Comment (JosiahWI) [inline include/cripts/Lulu.hpp:45]:
```suggestion
// Some convenience macros
```

Comment (zwoop) [reply include/cripts/Lulu.hpp:45]: Fixed

**Review by cmcfarlen — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**